### PR TITLE
Testing Ruby Code - Tic Tac Toe

### PR DESCRIPTION
**Pull request by tigamadou**
Specs ready to be merged to the developing branch

### CODE REVIEW

File: spec/board_spec.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+require_relative './../lib/board.rb'
+
+RSpec.describe Board do
+  describe '#available?' do
+    it 'return true if the move is available' do
+      board = Board.new
+      expect(board.available?(4)).to eql(true)
+    end
+  end
+end
```
Comment (ebukaume) [inline spec/board_spec.rb]: You should also test for a `false` case so you make sure you are not having a false positive.

File: spec/game_spec.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+game = Game.new(player1, player2)
+game.board = Board.new
+game2 = Game.new(player1, player2)
+game2.board = Board.new
+
+RSpec.describe Game do
+  let(:player1) { player1 }
+  let(:player2) { player2 }
+  let(:game) { game }
+
```
Comment (ebukaume) [inline spec/game_spec.rb]: Do these at the point you called `let`. Why did you have to do it that way?

File: spec/game_spec.rb [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      game.board.moves[1] = '0'
+      game.board.moves[2] = 'X'
+      expect(game.a_win?).to eql(false)
+    end
+  end
+
+  describe '#move_ok?' do
+    it 'return true if the move is available' do
+      expect(game.move_ok?(1)).to eql(false)
+    end
```
Comment (ebukaume) [inline spec/game_spec.rb]: What exactly are you testing here? Should `game.move_ok?(1)` not return `true`? If you use `let` properly, each instance of game in each example refers to a new game object.

File: spec/game_spec.rb [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    it 'return false if the move is not available' do
+      expect(game.move_ok?(10)).to eql(false)
+    end
+  end
+
+  describe '#turn' do
+    game.player = player1
```
Comment (ebukaume) [inline spec/game_spec.rb]: When you use `let` correctly as requested above, you would then move this into the `it` block to get it to work as expected.

File: spec/game_spec.rb [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    it 'Return Player2 if the current player is Player1' do
```
Comment (ebukaume) [inline spec/game_spec.rb]: This description is confusing. _It returns player2 if the current player is not player2_ :thinking:

File: spec/game_spec.rb [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+  describe '#draw?' do
+    game2.board.moves[0] = 'X'
+    game2.board.moves[1] = 'O'
+    game2.board.moves[2] = 'X'
+    game2.board.moves[3] = 'X'
+    game2.board.moves[4] = 'O'
+    game2.board.moves[5] = 'O'
+    game2.board.moves[6] = 'O'
+    game2.board.moves[7] = 'X'
+    game2.board.moves[8] = 'X'
```
Comment (ebukaume) [inline spec/game_spec.rb]: Remember to move this too into the `it` block after making the `let` fix.

**Review by ebukaume — CHANGES_REQUESTED** [review]
Hi @genzaraki and @Datagirlcmr 

Nice work so far. You guys are doing well; before we continue, let's make these changes below.

- [ ] You should test all the public methods in all your classes. [Ref](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/all_tic_tac_toe.png)

_See below for more_ :point_down: 

**Kindly submit another review request after making those changes**

Happy Coding :v: 
**Ebuka Umeokonkwo**
 [Github](https://github.com/ebukaume) | [Linkedin](https://www.linkedin.com/in/ebukaume) | [Slack](https://app.slack.com/team/UFDKXE68L) | [Twitter](https://twitter.com/ebukaume)

**Review by Redvanisation — CHANGES_REQUESTED** [review]
# Changes required  ♻️ 

### Hello team  🌟

**Well done implementing some of the changes, but we need to fix the issues below before moving on ⬇️** 

- Test all the public methods in your project both in positive and negative scenarios.

### _After making the required changes please submit a new code review request_

### Good luck! 🍀

**Review by imhta — APPROVED** [review]
# Project approved 🎉 

### Hello  @Datagirlcmr  @genzaraki , 👋 You have done a good job in this project 👏 



----
### Happy coding 😄 
----
**Pon Muthu Selvam N**
__nponmuthuselvam@gmail.com__

**Pull request merged**